torchvision/models/detection/transform.py

```diff
@@ -171,7 +171,12 @@ def resize_keypoints(keypoints, original_size, new_size):
 def resize_boxes(boxes, original_size, new_size):
     ratios = tuple(float(s) / float(s_orig) for s, s_orig in zip(new_size, original_size))
     ratio_height, ratio_width = ratios
-    xmin, ymin, xmax, ymax = boxes.unbind(1)
+    # unbind and lists not yet supported in ONNX
+    if torchvision._is_tracing():
+        xmin, ymin, xmax, ymax = [torch.squeeze(out, 1) for out in boxes.split(1, dim=1)]
```

Comment:
I'm sorry if I'm missing something here, but can't we implement `unbind` using `split` in ONNX?

By tracing
```python
In [35]: def f(x):
    ...:     xmin, ymin, xmax, ymax = [torch.squeeze(out, 1) for out in x.split(1, dim=1)]
    ...:     return xmin, ymin, xmax, ymax
    ...:

In [36]: ff = torch.jit.trace(f, torch.rand(3, 4))

In [38]: print(ff.graph_for(torch.rand(3, 4)))
graph(%0 : Float(*, *)):
  %1 : int = prim::Constant[value=1]() # /Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/tensor.py:311:0
  %3 : Tensor[] = aten::split(%0, %1, %1) # /Users/fmassa/anaconda3/lib/python3.6/site-packages/torch/tensor.py:311:0
  %out.1 : Tensor, %out.2 : Tensor, %out.3 : Tensor, %out : Tensor = prim::ListUnpack(%3)
  %9 : Tensor = aten::squeeze(%out.1, %1) # <ipython-input-35-2d3d6e2d1dd0>:2:0
  %11 : Tensor = aten::squeeze(%out.2, %1) # <ipython-input-35-2d3d6e2d1dd0>:2:0
  %13 : Tensor = aten::squeeze(%out.3, %1) # <ipython-input-35-2d3d6e2d1dd0>:2:0
  %15 : Tensor = aten::squeeze(%out, %1) # <ipython-input-35-2d3d6e2d1dd0>:2:0
  %16 : (Tensor, Tensor, Tensor, Tensor) = prim::TupleConstruct(%9, %11, %13, %15)
  return (%16)
```
we see that there is the use of `ListUnpack` and `TupleConstruct`.

From
```python
In [28]: def f(x):
    ...:     return x.unbind(0)
    ...:

In [29]: g = torch.jit.trace(f, torch.rand(3, 2))

In [31]: print(g.graph_for(torch.rand(3, 2)))
graph(%x : Float(*, *)):
  %1 : int = prim::Constant[value=0]() # <ipython-input-28-004b61e98d23>:2:0
  %2 : Tensor[] = aten::unbind(%x, %1) # <ipython-input-28-004b61e98d23>:2:0
  %3 : Tensor, %4 : Tensor, %5 : Tensor = prim::ListUnpack(%2)
  %6 : (Tensor, Tensor, Tensor) = prim::TupleConstruct(%3, %4, %5)
  return (%6)
```

we see that this is very similar to the graph that you currently have, and they both are using `ListUnpack` and `TupleConstruct`.

Thus, can't we just implement an override in PyTorch onnx to convert `unbind` to use `split` internally (or call `select` in a loop)?

Reply:
There is actually a PR from yesterday to support unbind https://github.com/pytorch/pytorch/pull/27247